http/src/main/resources/init-resources/gce-init.sh

```diff
@@ -464,10 +462,14 @@ if [ ! -z "$JUPYTER_DOCKER_IMAGE" ] ; then
   docker exec $JUPYTER_SERVER_NAME /bin/bash -c "R -e '1+1'" || true
 
   # For older jupyter images, jupyter_delocalize.py is using 127.0.0.1 as welder's url, which won't work now that we're no longer using `network_mode: host` for GCE VMs
-  docker exec $JUPYTER_SERVER_NAME sed -i  "s/127.0.0.1/welder/g" /etc/jupyter/custom/jupyter_delocalize.py
+  docker exec $JUPYTER_SERVER_NAME /bin/bash -c "sed -i 's/127.0.0.1/welder/g' /etc/jupyter/custom/jupyter_delocalize.py"
+
+  # make sure home directory is owned by jupyter-user
+  docker exec -u root $JUPYTER_SERVER_NAME /bin/bash -c "chown -R jupyter-user:users ${NOTEBOOKS_DIR}" || true
+  docker exec -u root $JUPYTER_SERVER_NAME /bin/bash -c "chown -R jupyter-user:users ${JUPYTER_USER_HOME}/.local/share/jupyter/" || true
 
   log 'Starting Jupyter Notebook...'
-  retry 3 docker exec -d ${JUPYTER_SERVER_NAME} ${JUPYTER_SCRIPTS}/run-jupyter.sh ${NOTEBOOKS_DIR}
+  retry 3 docker exec -u jupyter-user -d $JUPYTER_SERVER_NAME /bin/bash -c "${JUPYTER_SCRIPTS}/run-jupyter.sh ${NOTEBOOKS_DIR}"
```

Comment:
Ah I forgot the user is different in new images. I don't think `-u` is really needed, I was trying to fix test flakiness (thought it might be starting as root somehow)

Reply:
Probably also need to fix the `chown` commands above (again, not sure that is really needed, was trying to fix test flakiness but wasn't really successful).